src/llm/http_llm_calculator.cc

```diff
@@ -108,11 +116,21 @@ class OpenAIChatCompletionsRequest {
     }
 
     chat_t getMessages() const { return this->messages; }
+    Endpoint getEndpoint() const { return this->endpoint; }
+    void setEndpoint(Endpoint endpoint) { this->endpoint = endpoint; }
+    std::string getPrompt() const {
+        if (this->prompt.has_value()) {
+            return this->prompt.value();
+        } else {
+            return "";
+        }
+    }
+
     bool isStream() const { return this->stream; }
     std::string getModel() const { return this->model; }
 
-    // TODO: Use exceptions to sneak error mesages into response
-    bool parse() {
+    // TODO: Use exceptions to sneak error messages into response
+    bool parse(Endpoint endpoint) {
```

Comment:
There is no need to pass endpoint to `parse`, it is already a member of `OpenAIChatCompletionsRequest ` class